xdsl/dialects/fsm.py

```diff
@@ -297,20 +299,22 @@ def verify_(self) -> None:
             raise VerifyException("Destination is not a variable operation")
 
         parent = self.parent_op()
-        assert isinstance(parent, TransitionOp)
-
-        found = 0
-        for op in parent.action.walk():
-            if isinstance(op, UpdateOp) and op.variable == self.variable:
-                found += 1
-        if found == 0:
-            raise VerifyException(
-                "Update must only be located in the action region of a transition"
-            )
-        elif found > 1:
-            raise VerifyException(
-                "Multiple updates to the same variable within a single action region is disallowed"
-            )
+        while parent is not None:
+            if isinstance(parent, TransitionOp):
+                # walk through the action region
+                found = 0
+                for op in parent.action.walk():
+                    if isinstance(op, UpdateOp) and op.variable == self.variable:
+                        found += 1
+                if found == 0:
+                    raise VerifyException(
+                        "Update must only be located in the action region of a transition"
+                    )
+                elif found > 1:
+                    raise VerifyException(
+                        "Multiple updates to the same variable within a single action region is disallowed"
+                    )
+            parent = parent.parent_op()
```

Comment:
I feel that this change is backwards, maybe it's a merge problem?
If you want I could take a look with you tomorrow

Reply:
yup I guess you're right, I had quite some issues with main merging-rebasing so I guess I missed that! Sure thing :)